packages/server/src/oauth/middleware.ts

```diff
@@ -76,7 +87,34 @@ async function authenticateBasicAuth(req: IncomingMessage, token: string): Promi
     authTime: new Date().toISOString(),
   };
 
-  return { login, project, membership };
+  const onBehalfOfHeader = req.headers['x-medplum-on-behalf-of'];
+  let onBehalfOf: ProfileResource | undefined = undefined;
+  let onBehalfOfMembership: ProjectMembership | undefined = undefined;
+
+  if (onBehalfOfHeader && isString(onBehalfOfHeader)) {
+    if (!membership.admin) {
+      throw new OperationOutcomeError(unauthorized);
+    }
+
+    if (onBehalfOfHeader.startsWith('ProjectMembership/')) {
+      onBehalfOfMembership = await systemRepo.readReference<ProjectMembership>({ reference: onBehalfOfHeader });
+    } else {
+      onBehalfOfMembership = await systemRepo.searchOne({
+        resourceType: 'ProjectMembership',
+        filters: [
+          { code: 'profile', operator: Operator.EQUALS, value: onBehalfOfHeader },
+          { code: 'project', operator: Operator.EQUALS, value: getReferenceString(project) },
+        ],
+      });
```

Comment:
out of curiosity, why do we need system repo here? We're only looking for resources scoped to the current project, and the caller should be an admin, so they should have access to `ProjectMemberships`. 

Would save us from having to remember to filter on `project`

Reply:
Good suggestion, and that would work in theory.  However, this is in the middle of the auth flow.  We don't have a repo for the current user yet, because this is the code where we're setting up the current user.

We could create an intermediate `Repository` using `getRepoForLogin` yada yada, but it would be a bunch of ceremony for what logically boils down to the current code.

Reply:
Thought about this more, and I think you're right that the extra ceremony might be worth it, especially when thinking about future proofing for cases where project admins might have access policies that restrict which users they can see.